messaging/src/main/java/org/axonframework/queryhandling/distributed/DistributedQueryBus.java

```diff
@@ -197,16 +197,7 @@ public MessageStream<QueryResponseMessage> query(@Nonnull QueryMessage query) {
             queryingExecutor.execute(
                     new PriorityRunnable(() -> {
                         try {
-                            var result = localSegment.query(query, null);
-                            result.first()
-                                  .asCompletableFuture()
-                                  .whenComplete((firstMessage, error) -> {
-                                      if (error != null) {
-                                          localResult.completeExceptionally(error);
```

Comment:
Question:
Why do we complete exceptionally in case of exception throw by the `localSegment.query(query, null)`, but not in the case of the error in the resulting MessageStream?

Reply:
It's what `DelayedMessageStream.create(localResult)` takes care of. There is no need to read anything from that stream.

Reply:
Thanks, I've understood it from the previous response :)